extensions/src/test/java/org/pentaho/platform/plugin/services/email/EmailServiceTest.java

```diff
@@ -0,0 +1,285 @@
+/*!
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ *
+ * Copyright (c) 2002-2023 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.platform.plugin.services.email;
+
+import junit.framework.TestCase;
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.http.HttpException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.pentaho.di.core.encryption.Encr;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Properties;
+
+@RunWith( MockitoJUnitRunner.class )
+public class EmailServiceTest extends TestCase {
+
+  private EmailService emailService = null;
+
+  private File defaultConfigFile = null;
+  @Before
+  public void setUp() throws Exception {
+    defaultConfigFile = File.createTempFile( "email_config_", ".xml" );
+    this.emailService = new EmailService( defaultConfigFile );
+  }
+
+
+  @Test( expected = HttpException.class )
+  public void testGetOAuthToken() throws HttpException, IOException {
+    EmailConfiguration emailConfigOriginal =
+            new EmailConfiguration( true, false, "tes@test.com", "Pentaho Scheduler", "", 25,
+                    "graph_api", true, "test@test.com", "", true, true );
+    emailConfigOriginal.setAuthMechanism( "XOAUTH2" );
+    emailConfigOriginal.setClientId( "cid" );
+    emailConfigOriginal.setClientSecret( "secret" );
+    emailConfigOriginal.setTokenUrl( "https://login.microsoftonline.com/common/oauth2/v2.0/token" );
+    emailConfigOriginal.setScope( "https://graph.microsoft.com/.default" );
+    emailConfigOriginal.setGrantType( "client_credentials" );
+    emailService.getOAuthToken( emailConfigOriginal );
```

Comment:
not a must have.

But add a java comment to briefly explain why an exception should be thrown here